ecl/eclcmd/eclcmd_core.cpp

```diff
@@ -1959,15 +1964,38 @@ class EclCmdZapGen : public EclCmdCommon
             if ((strstr(errMsg.str(), "Warning") != nullptr) || (strstr(errMsg.str(), "Failed") != nullptr))
             {
                 fprintf(stderr, "Zapgen command returns with error:\n%s\n", errMsg.str());
-                return false;
+                isOk = false;
+            }
+            else if ((strstr(errMsg.str(), "< Location") != nullptr) && (strstr(errMsg.str(), "Login.html") != nullptr))
```

Comment:
Would be nice if this were a more explicit error message being checked... is this output by ESP?  Is it the attempt to redirect to the login screen?

@wangkx is there a way for us to know that this is a REST client rather than someone that needs to go to the login page?

Reply:
Based on the curl report it seems a redirection attempt.

Reply:
We do not have a way to detect a REST client. We are using the following rule to decide whether ESP should redirect a client to login page or simply send a 401:
bool rLoginPage = authorizationHeader.isEmpty() && !authReq.authBinding->isCORSRequest(originHeader.str()) &&
        !strieq(authReq.httpMethod.str(), POST_METHOD));

Reply:
So, 1. @AttilaVamos if the user had provided authentication are you adding it through CURL?
      2. @wangkx if auth is there it won't redirct?
      3. @AttilaVamos , can you send this as a POST instead of a GET?
      4. @wangkx if its a POST it won't redirect?

Reply:
I am not sure that I understand the function here. For session enabled ESP, if a user is authenticated, ESP will not redirect to a login page. If a user has not authenticated, ESP will not redirect to the login page if the request is a POST.